### PR TITLE
bugfix: when service works with upstream that contains a domain name, the merged configuration should not always changing.

### PR DESCRIPTION
**Pull request by membphis**
### What this PR does / why we need it:

when service works with upstream that contains a domain name, the merged configuration is always changing.
if the configuration is always changing,  it won't hit the LRU item, which causes some plugins to work abnormally, such as `limit-req`.

fix: https://github.com/apache/apisix/issues/2320

### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

File: t/plugin/key-auth-upstream-domain-node.t [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+                        "limit-req": {
+                            "rate": 1,
+                            "burst": 0,
+                            "rejected_code": 503,
+                            "key": "remote_addr"
+                        }
+                    },
+                    "upstream": {
+                        "nodes": {
+                            "www.baidu.com:80": 1
```
Comment (nic-chen) [inline t/plugin/key-auth-upstream-domain-node.t:100]: seems visit `baidu.com` unstable in github actions before?

File: apisix/init.lua [hunk 2]
```diff
@@ -461,23 +455,15 @@ function _M.http_access_phase()
 
     else
         if route.has_domain then
-            local parsed_route, err = lru_resolved_domain(route, api_ctx.conf_version,
-                                        return_direct, nil)
+            local err
+            route, err = lru_resolved_domain(route, api_ctx.conf_version,
+                            parse_domain_in_route, route, api_ctx)
```
Comment (gxthrj) [inline apisix/init.lua:460]: `api_ctx` need to be removed ？

**Review by gxthrj — APPROVED** [review]

File: apisix/init.lua (hunk 2 above)
Comment (membphis) [reply apisix/init.lua:460]: nice, will fix it later

**Comment by membphis** [comment]
@moonming this is an old bug for long time, and it seems that I need more time to fix it. 
Therefore, I think it is not necessary for the 2.0 milestone.

**Comment by moonming** [comment]
ok

**Review by nic-chen — APPROVED** [review]

**Review by spacewander — APPROVED** [review]

**Pull request merged**